src/main/java/entities/Topic.java

```diff
@@ -0,0 +1,71 @@
+package entities;
+
+import net.dv8tion.jda.api.entities.Member;
+
+import java.util.LinkedList;
+
+/**
+ * A topic for a server. Internally contains a queue of Members.
+ */
+public class Topic {
+    private final String name;
+    private final LinkedList<Member> queue = new LinkedList<>();
+
+    /**
+     * Constructs a new Topic object. This does not automatically create
+     * the topic on the Discord server.
+     * @param name The name of the topic.
+     */
+    public Topic(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Add a Member to the back of the queue.
+     * @param member The Member to add
+     */
+    public void addToQueue(Member member) {
+        queue.add(member);
+    }
+
+    /**
+     * Remove a Member from their position in the queue.
+     * @param member The Member to remove
+     */
+    public void removeFromQueue(Member member) {
+        queue.remove(member);
+    }
+
+    /**
+     * Check if a Member is inside the queue.
+     * @param member The Member to check
+     * @return True if the member is in the queue, false otherwise
+     */
+    public boolean isInQueue(Member member) {
+        return queue.contains(member);
+    }
+
+    /**
+     * Returns an array of all Members in the queue.
+     * @return The Members in this queue
+     */
+    public Member[] getMembersInQueue() {
+        return queue.toArray(new Member[0]);
```

Comment:
That's what I thought, but IntelliJ gave me a warning saying that this was a better alternative. I think `Collection.toArray()` automatically creates an array of the proper size if the existing one is too small, so this works.

Reply:
IntelliJ's explanation:

> There are two styles to convert a collection to an array: either using a pre-sized array (like c.toArray(new String[c.size()])) or using an empty array (like c.toArray(new String[0]). In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation. This inspection allows to follow the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs).

Reply:
I love these paragraph explanations! Could you add a small comment summing that up for future ref?

Reply:
Yeah, I'll add a comment here.